app/main/views/suppliers.py

```diff
@@ -263,6 +264,56 @@ def edit_supplier_organisation_size():
     return render_template('suppliers/edit_supplier_organisation_size.html', form=form)
 
 
+@main.route('/vat-number/edit', methods=['GET', 'POST'])
+@login_required
+def edit_supplier_vat_number():
+    form = VatNumberForm()
+    try:
+        supplier = data_api_client.get_supplier(current_user.supplier_id)['suppliers']
+    except APIError as e:
+        abort(e.status_code)
+
+    if supplier.get("vatNumber"):
+        return (
+            render_template("suppliers/already_completed.html", completed_data_description="vat number"),
+            200 if request.method == 'GET' else 400
+        )
+
+    form_errors = None
+    if request.method == 'POST':
+        if form.validate_on_submit():
+            maybe_vat_number = form.vat_number.data if form.vat_registered.data == 'Yes' else 'Not VAT registered'
+
+            try:
+                data_api_client.update_supplier(supplier_id=current_user.supplier_id,
+                                                supplier={"vatNumber": maybe_vat_number},
+                                                user=current_user.email_address)
+            except APIError as e:
+                abort(e.status_code)
+
+            return redirect(url_for('.supplier_details'))
+
+        current_app.logger.warning(
```

Comment:
What is this logging about?

Reply:
It's just giving handy feedback on why something might potentially fail. It seems to be a pattern we do for supplier account views, just continuing with it really.

Reply:
Ahh, ok, thank you ^_^